modules/security-cors/src/main/java/com/devonfw/module/security/cors/common/impl/WebSecurityConfigurerAspectCors.java

```diff
@@ -0,0 +1,48 @@
+package com.devonfw.module.security.cors.common.impl;
+
+import javax.inject.Inject;
+
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.web.cors.CorsConfiguration;
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
+import org.springframework.web.filter.CorsFilter;
+
+import com.devonfw.module.security.common.api.config.WebSecurityConfigurerAspect;
+
+/**
+ * Implementation of {@link WebSecurityConfigurerAspect} for CORS protection.
+ *
+ * @since 2020.12.001
+ */
+public class WebSecurityConfigurerAspectCors implements WebSecurityConfigurerAspect {
+
+  private CorsConfigProperties corsConfigProperties;
+
+  /**
+   * @param corsConfigProperties new value of {@link #getcorsConfigProperties}.
+   */
+  @Inject
+  public void setCorsConfigProperties(CorsConfigProperties corsConfigProperties) {
+
+    this.corsConfigProperties = corsConfigProperties;
+  }
+
+  private CorsFilter getCorsFilter() {
+
+    UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+    CorsConfiguration config = new CorsConfiguration();
+    config.setAllowCredentials(this.corsConfigProperties.isAllowCredentials());
+    config.setAllowedOrigins(this.corsConfigProperties.getAllowedOrigins());
+    config.setAllowedHeaders(this.corsConfigProperties.getAllowedHeaders());
+    config.setAllowedMethods(this.corsConfigProperties.getAllowedMethods());
```

Comment:
All this mapping will not be needed anymore with my suggested improvement as you can use `this.corsConfigProperties` and directly pass it to `source`.